apps/src/aichat/types/miscellaneous.ts

```diff
@@ -0,0 +1,29 @@
+import {ValueOf} from '@cdo/apps/types/utils';
+import {AiChatTeacherFeedback} from '@cdo/generated-scripts/sharedConstants';
+
+import {ChatMessage} from './chatEvents';
+
+// Miscellaneous types used through the AI Chat lab.
+// If possible, try to create new types in more specific files.
```

Comment:
Just sharing a couple thoughts here as I started looking at your other refactoring PR https://github.com/code-dot-org/code-dot-org/pull/63775. Sorry late to the party!

- `ChatCompletionAPIResponse` and `AichatContext` could be placed in a new `chatCompletion` category.
- Since `ChatEventDescriptionKey` is already included in `chatEvents`, we could place `FeedbackValue` in there as well.
- `SaveType` could be moved to `customizations` (saving the part of model customizations and maybe `ViewMode` seems like it could be moved there as well since we can either edit customizations or view information about them.

Reply:
Yep good points! Will continue to clean these up in future PRs. I already have a change prepped that'll get rid of the `ChatCompletionApiResponse` type

Reply:
Oh nice! I was wondering about that when look at the miscellaneous types since we removed `ChatEventApiResponse`. Thanks!